src/tonic/api/collections_api.rs

```diff
@@ -166,6 +170,102 @@ impl Collections for CollectionsService {
         validate(request.get_ref())?;
         self.list_aliases(request).await
     }
+
+    async fn collection_cluster_info(
+        &self,
+        request: Request<CollectionClusterInfoRequest>,
+    ) -> Result<Response<CollectionClusterInfoResponse>, Status> {
+        validate(request.get_ref())?;
+        let cluster_info = do_get_collection_cluster(
+            self.dispatcher.toc(),
+            request.into_inner().collection_name.as_str(),
+        )
+        .await
+        .map_err(|e| Status::new(tonic::Code::Unknown, e.to_string()))?;
+
+        let response = CollectionClusterInfoResponse {
+            peer_id: cluster_info.peer_id,
+            shard_count: cluster_info.shard_count as u64,
+            local_shards: cluster_info
+                .local_shards
+                .iter()
+                .map(|shard| LocalShardInfo {
```

Comment:
we usually place the `gRPC` object conversions [here](https://github.com/qdrant/qdrant/blob/master/lib/api/src/grpc/conversions.rs)

Reply:
Good catch :+1: 

Sorry I picked the wrong `conversions.rs` file, happy you were able to figure this on your own.